android/src/main/kotlin/com/adyen/checkout/flutter/components/action/ActionComponentManager.kt

```diff
@@ -0,0 +1,58 @@
+package com.adyen.checkout.flutter.components.action
+
+import ActionComponentConfigurationDTO
+import ComponentFlutterInterface
+import androidx.fragment.app.FragmentActivity
+import com.adyen.checkout.action.core.GenericActionComponent
+import com.adyen.checkout.components.core.ActionComponentCallback
+import com.adyen.checkout.components.core.action.Action
+import com.adyen.checkout.flutter.components.view.ComponentLoadingBottomSheet
+import com.adyen.checkout.flutter.utils.ConfigurationMapper.mapToCheckoutConfiguration
+import org.json.JSONObject
+import java.util.UUID
+
+class ActionComponentManager(
+    private val activity: FragmentActivity,
+    private val componentFlutterApi: ComponentFlutterInterface,
+) {
+    private var actionComponent: GenericActionComponent? = null
+    private var componentId: String? = null
+
+    fun createActionComponent(
+        actionComponentConfigurationDTO: ActionComponentConfigurationDTO,
+        componentId: String
+    ): GenericActionComponent {
+        val checkoutConfiguration = actionComponentConfigurationDTO.mapToCheckoutConfiguration()
+        val genericActionComponent =
+            GenericActionComponent.PROVIDER.get(
+                activity,
+                checkoutConfiguration,
+                ActionComponentCallback(
+                    activity,
+                    componentFlutterApi,
+                    componentId,
+                ),
+                UUID.randomUUID().toString()
```

Comment:
Will this cause a new component to be created when you rotate the screen for example?